src/lib/hash-links/use-active-section.ts

```diff
@@ -15,17 +14,22 @@ import getFullNavHeaderHeight from 'lib/get-full-nav-header-height'
  * https://github.com/hashicorp/react-components/pull/325
  */
 export function useActiveSection(
-	headings: TableOfContentsHeading[],
-	isEnabled = true
+	slugs: string[],
+	isEnabled: boolean = true
 ): string {
 	const visibleHeadings = useRef<Set<string>>(new Set())
-	const [activeSection, setActiveSection] =
-		useState<TableOfContentsHeading['slug']>()
+	const [activeSection, setActiveSection] = useState<string>()
 	const previousY = useRef<number>()
 
-	// isProductLanding is needed to determine the IntersectionObserver threshold
-	// because the headings on product landing pages are smaller than the on docs
-	// pages
+	/**
+	 * isProductLanding is needed to determine the IntersectionObserver threshold
+	 * because the headings on product landing pages are smaller than the on docs
+	 * pages
+	 *
+	 * TODO: consider extracting this as a more generic option?
+	 * For example, support an optional `observerThreshold` argument, default 1?
+	 * And pass the option in when using this component on product landing pages?
+	 */
```

Comment:
💭 I'm actually working on making all the headings the same size in the consolidation side work (#1634), so we should be able to remove this logic after that's in soon!

Reply:
I see the comment you added! Thanks a bunch 😊